R/particle_filter.R

```diff
@@ -458,15 +458,24 @@ particle_resample <- function(weights) {
 ## Private helper for reconstituting a particle filter from its
 ## `$inputs()` data, but possibly changing the seed
 particle_filter_from_inputs <- function(inputs, seed = NULL) {
-  particle_filter$new(data = inputs$data,
-                      model = inputs$model,
-                      n_particles = inputs$n_particles,
-                      compare = inputs$compare,
-                      device_config = inputs$device_config,
-                      index = inputs$index,
-                      initial = inputs$initial,
-                      n_threads = inputs$n_threads,
-                      seed = seed %||% inputs$seed)
+  if (is.null(inputs$n_particles)) {
```

Comment:
Again, this part is an implementation detail and not something exposed to the user.

If you ran the normal particle filter with 1 particle it would be roughly equivalent but slower, yes. The `NULL` value here has a special meaning in dust though, as it puts it into "multiparameter" mode, which affects the way output is structured